### PR TITLE
Silver anaphora and relation annotations

### PR DESCRIPTION
**Pull request by ringgaard**
* I have added a simple anaphora annotator for the silver data generation. Only 3rd person singular gendered personal pronouns are resolved in English and Danish so far. It also has limited support for definite references to the topic entity.
* Relation annotator for silver data generation. This finds pairs of mentions of entities in sentences that can be related using a simple fact from the knowledge base.
* Bold intro phrase annotator.
* Black-listed properties for phrase annotator.

Minor changes:
* Switch to using bazel v1.0 for Travis.
* Add numeric dates as facts in fact extractor.
* Take hyper-threads into account when determining the number of CPUs.

### CODE REVIEW

File: third_party/jit/cpu.cc [hunk 1]
```diff
@@ -330,7 +330,21 @@ void CPU::Initialize() {
...
 int CPU::Processors() {
-  return get_nprocs();
+  static int cpus = -1;
+  if (cpus == -1) {
+    // Get the number of processors.
+    int processors = get_nprocs();
+
+    // Check if hyper-treading is enabled.
+    bool htt = false;
+    FILE *f = fopen("/sys/devices/system/cpu/smt/active", "r");
```
Comment (anders-sandholm) [inline third_party/jit/cpu.cc]: Can't help but smile looking at this code :-)

File: sling/nlp/ner/anaphora.cc [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+        t++;
+      }
+    }
+  }
+
+ private:
+  // Pronoun gender.
+  enum Gender {UNKNOWN, MASCULINE, FEMININE, NEUTRAL};
+
+  // Pronoun descriptor. This also convers trigger words that might not
```
[the remark this answers is not in the document]
Comment (ringgaard) [reply sling/nlp/ner/anaphora.cc]: Fixed

File: sling/nlp/ner/annotators.cc [hunk 3]
```diff
@@ -207,6 +207,43 @@ void EmphasisAnnotator::Annotate(SpanChart *chart) {
...
+    if (chart->token(b).style() & BOLD_BEGIN) {
+      begin = b;
+      break;
+    }
+  }
+  if (begin == -1) return;
+
+  // Find end of bolded span.
+  int end = -1;
+  for (int e = begin + 1; e < chart->size() && e - begin < max_length; ++e) {
```
[the remark this answers is not in the document]
Comment (ringgaard) [reply sling/nlp/ner/annotators.cc]: I could do that, but the line would be longer than 80 chars, so it doesn't look very pretty.

File: third_party/jit/cpu.cc (hunk 1 above)
Comment (ringgaard) [reply third_party/jit/cpu.cc]: It is a bit hacky, but it does the job.

**Pull request merged**